spec/term/1.0.md

```diff
@@ -0,0 +1,127 @@
+
+### 1.1. Terminals
+
+When a VT6 server agrees to using this module, it MUST have a terminal that its clients can interact with as described in this specification.
+
+A **terminal** is a triple of:
+- a two-dimensional array of Unicode characters,
+- a **cursor** that points at a position in this character array, and
+- an input stream from which Unicode characters can be read.
+
+*Rationale:* A sole printer can provide such a terminal even though it has no input, because the definition doesn't require that the input stream ever becomes ready for reading.
+
+The character array is made up of **columns** of characters along the **horizontal** dimension, and of **rows** of characters along the **vertical** dimension.
+The number of columns and rows in the character array is called the terminal's **width** and **scrollback height**, respectively.
+
+Columns are numbered from left to right, and rows are numbered from top to bottom, in both cases starting with zero.
+A position in the array is addressed by a pair of the index of its row and the index of its column.
+For example, the position `(0,0)` refers to the character at the top left corner of the array.
+
+For graphical terminals, the character array MAY be much larger than what can be displayed on screen at once.
+In this case, the character array itself is called the terminal's **scrollback**, and the range of currently visible rows is called the terminal's **viewport**.
+The number of rows in the viewport is called the terminal's **viewport height**.
+
+*Rationale:* The term "terminal height" is not defined to avoid confusion. We always refer explicitly to either the scrollback height or the viewport height.
+
+### 1.2. Standard output
+
+When a client writes a byte string into its standard output, the server SHALL assume that this byte string is encoded in [UTF-8](https://tools.ietf.org/html/rfc3629), and decode it into a string of Unicode characters.
+
+When the `term1.output-protected` property is true, the server MUST discard all characters from this string that are in the general categories Cc ("Other, control"), Cf ("Other, format"), Cs ("Other, surrogate"), Co ("Other, private use") and Cn ("Other, not assigned"), except for U+0009 (Horizontal Tab), U+000A (New Line) and U+000D (Carriage Return).
+
+When the `term1.output-protected` property is false, the server MAY alternatively:
+
+- choose a printable representation for each such character and replace the character with that representation before continuing, or
+- interpret the character as starting a control sequence if the character is U+001B (Escape).
+
+When a control character is interpreted as starting a control sequence, the entire control sequence SHALL be removed from the string before processing as described in this section continues, and the remaining string MAY be modified depending on the semantics of the control sequence.
```

Comment:
Need to add a requirement that ISO 2022 sequences MUST NOT be interpreted under any circumstances. *Rationale:* It's the 21st century.